src/aws/flb_aws_util.c

```diff
@@ -322,15 +328,46 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
         flb_warn("[aws_http_client] failed to increase max response buffer size");
     } 
 
-    /* Add AWS Fluent Bit user agent */
-    if (aws_client->extra_user_agent == NULL) {
+    /* 
+    * Set AWS Fluent Bit user agent FLB_AWS_USER_AGENT
+    * This environment variable needs to be set when:
+    * 1. it is not set;
+    * 2. extra_user_agent has been specified by one aws_client and 
+    * its value doesn't match current AWS Fluent Bit user agent
+    */
```

Comment:
the 2nd should be reversed right... we don't use FLB_AWS_USER_AGENT when extra_user_agent is already set I htought?

Reply:
yeah so the problem is that based on my test, even we set extra_user_agent in CW config, it is not set at the very beginning. Fluent Bit will call this file first (without CW client triggered) and then trigger CW (with extra_user_agent set)